lib/media/track/localaudiotrack.js

```diff
@@ -80,10 +113,50 @@ class LocalAudioTrack extends LocalMediaAudioTrack {
    * @fires LocalAudioTrack#stopped
    */
   stop() {
+    this._didCallStop = true;
     return super.stop.apply(this, arguments);
   }
 }
 
+function restartWhenInadvertentlyStopped(localAudioTrack) {
+  const { _getUserMedia: getUserMedia, _log: log } = localAudioTrack;
+
+  function shouldReacquireTrack() {
+    return document.visibilityState === 'visible'
+      && (localAudioTrack.mediaStreamTrack.muted || localAudioTrack.isStopped)
+      && !localAudioTrack._didCallStop;
+  }
+
+  function reacquireTrack(constraints) {
+    return getUserMedia({ audio: constraints, video: false }).then(mediaStream => {
+      return mediaStream.getAudioTracks()[0];
+    });
+  }
+
+  function handleTrackStateChange() {
+    Promise.race([
+      waitForEvent(localAudioTrack.mediaStreamTrack, 'unmute'),
+      waitForSometime(50)
+    ]).then(() => {
+      log.debug('checking if audio track needs to be replaced');
+      if (!shouldReacquireTrack()) {
+        return;
+      }
+      const constraints = localAudioTrack.mediaStreamTrack.getConstraints();
+      reacquireTrack(constraints).then(newTrack => {
+        log.debug('replacing track with constrains ', newTrack.id, constraints);
+        return localAudioTrack._setMediaStreamTrack(newTrack);
+      }).catch(err => {
+        log.debug('Failed to acquire track', err);
+      });
+    });
```

Comment:
Please modify the log statements:
```suggestion
    ]).then(() => {
      if (!shouldReacquireTrack()) {
        return;
      }
      const constraints = localAudioTrack.mediaStreamTrack.getConstraints();
      log.info('Re-acquiring the MediaStreamTrack.');
      log.debug('Constraints:', constraints);
      return reacquireTrack(constraints).then(newTrack => {
        log.info('Re-acquired the MediaStreamTrack.');
        log.debug('MediaStreamTrack:', newTrack);
        return localAudioTrack._setMediaStreamTrack(newTrack);
      }).catch(err => {
        log.warn('Failed to re-acquire the MediaStreamTrack:', err);
      });
    });
```